addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -124,12 +117,34 @@ def post(self):
                 raise UserError(_('We do not accept the usage of document types on receipts yet. '))
         return super().post()
 
-    @api.constrains('name', 'journal_id', 'state')
-    def _check_unique_sequence_number(self):
-        """ This uniqueness verification is only valid for customer invoices, and vendor bills that does not use
-        documents. A new constraint method _check_unique_vendor_number has been created just for validate for this purpose """
-        vendor = self.filtered(lambda x: x.is_purchase_document() and x.l10n_latam_use_documents)
-        return super(AccountMove, self - vendor)._check_unique_sequence_number()
+    @api.constrains('ref', 'move_type', 'partner_id', 'journal_id', 'invoice_date')
+    def _check_duplicate_supplier_reference(self):
```

Comment:
@william-andre I am a bit surprised we would still need this.

Reply:
It wasn't the case before but AL made us add it because some vendors reuse the same number a  lot (they have only 100 numbers that get reused all the time for instance; use case in Belgium)